R/utils.R

```diff
@@ -324,6 +324,145 @@ date_to_numeric_modulus <- function(dt, date_column, timestep) {
   return(dt[])
 }
 
+#' Set caching location for Stan models
+#'
+#' This function allows the user to set a cache location for
+#' Stan models rather than a temp directory. This can reduce the
+#' need for model compilation on every new model run across sessions.
+#'
+#' @param path A valid filepath representing the desired cache location
+#'
+#' @return The string of the filepath set
+#' @examples
+#' # Set to local directory
+#' dir.create("test")
+#' enw_set_cache("test")
+#' enw_get_cache()
+#' # Use  the package cache in R >= 4.0
+#' if(R.version.string >= 4.0) {
+#'   enw_set_cache(tools::R_user_dir(package = "epinowcast", "cache"))
+#' }
+#' @importFrom cli cli_abort cli_alert
+#' @export
+
+enw_set_cache <- function(path = NULL) {
+
+  if (is.null(path)) {
+    cli::cli_abort("`path` must be a valid file path.")
+  }
+
+  prior_cache <- Sys.getenv("enw_cache_location", unset = "", names = NA)
+
+  if (!check_environment_setting(prior_cache)) {
+    cli::cli_alert("{prior_cache} exists and will be overwritten")
+  }
+  env_contents_active <- enw_get_environment_contents()
+
+  #!TODO should we normalise the path?
+  candidate_path <- normalizePath(path, winslash = "\\", mustWork = FALSE)
+
+  enw_environment <- paste0("enw_cache_location=\"", candidate_path, "\"\n")
+
+  new_env_contents <- append(
+    env_contents_active[["env_contents"]],
+    enw_environment
+  )
+
+  writeLines(
+    new_env_contents,
+    con = env_contents_active[["env_path"]], sep = "\n"
+  )
+
+  readRenviron(env_contents_active[["env_path"]])
+
+  invisible(candidate_path)
+
+}
+
+#' Unset Stan cache location
+#'
+#' Removes `enw_cache_location` environment variable from
+#' the user .Renviron file and removes it from the local
+#' environment.
+#'
+#' @return the prior cache location, if it existed
+#'
+#' @examples
+#' enw_unset_cache()
+#'
+#' @family utils
+#' @export
+
+enw_unset_cache <- function() {
+  prior_location <- Sys.getenv("enw_cache_location")
+
+  Sys.unsetenv("enw_cache_location")
+
+  invisible(enw_get_environment_contents(remove_enw_cache_location = TRUE))
+
+  return(invisible(prior_location))
+
+}
+
+
+#' Retrieve Stan cache location
+#'
+#' Retrieves the user set cache location for Stan models. This
+#' path can be set through the `enw_cache_location` function call.
+#' If no environmental variable is available the output from
+#' `tempdir` will be returned.
+#'
+#' @return The string of the filepath
+#'
+#' @return a string representing the file path for the cache location
+#'
+#' @importFrom cli cli_inform
+#'
+#' @family utils
+#' @export
+
+enw_get_cache <- function() {
+  cache_location <- Sys.getenv("enw_cache_location")
+
+  if (is.null(cache_location) || cache_location == "") {
+    cache_location <- tempdir()
```

Comment:
Do you think we should have a cli message here to flag using the temporary directory?

Reply:
Agree.